README.md

```diff
@@ -154,6 +154,69 @@ IResult<string> result.Map(value => value.toString());
 
 If `result` represents an failure all calls to `Map` and `FlatMap` are ignored. See `FaunaDB.Types.Result`.
 
+### How to work with user defined classes
+
+Instead of create manually your objects using our dsl (via `Obj()` method) it's possible to use our `Encoder` class and convert any user defined class to an equivalent `Value` type.
+
+Consider the following `Product` class:
+
+```csharp
+class Product
+{
+    [FaunaField("description")]
+    public string Description { get; set; }
+
+    [FaunaField("price")]
+    public double Price { get; set; }
+
+    [FaunaConstructor]
+    public Product(string description, double price)
+    {
+        Description = description;
+        Price = price;
+    }
+}
+```
+
+It's possible to create and persist an instance of `Product` on FaunaDB using:
+
+```csharp
+Product product = new Product("Smartphone", 649.90);
+
+await client.Query(
+    Create(
+        Ref("classes/product"),
+        Obj("data", Encoder.Encode(product))
+    )
+);
+```
+
+Also it's possible to convert back from a `Value` type to a `Product` type using our `Decoder` class:
+
+```csharp
+Value value = await client.Query(Get(Ref("classes/product/123456789")));
+
+Product product = Decoder.Decode<Product>(value);
+```
+
+or using a sugar method `To<T>()` like:
+
+```csharp
+Value value = await client.Query(Get(Ref("classes/product/123456789")));
+
+IResult<Product> product = value.To<Product>();
+```
+
+In that case it'll return a `IResult<T>` type.
+
+Those are three attributes that can be used to change the behavior of the `Encoder` and `Decoder`:
+
+- `FaunaField`: Used to override a custom field name and/or provide a default value for that field. If this attribute is not specified, the member name will be used instead. Can be used on fields, properties and constructors arguments.
+- `FaunaConstructor`: Used to mark a constructor or a public static method as an instantiator of the specified type. This attribute can be used only once per class.
```

Comment:
I think its clearer to change "as an instantiator of the specified type" => "as the method used to instantiate the specified type"